signac/core/syncedattrdict.py

```diff
@@ -35,7 +35,8 @@ class SyncedAttrDict(SyncedCollection, MutableMapping):
         dictionary representation, and if necessary construct a new SyncedAttrDict.
     """
 
-    _PROTECTED_KEYS = ('_data', '_suspend_sync_', '_load', '_sync', '_parent')
+    _PROTECTED_KEYS = ('_data', '_suspend_sync_', '_load', '_sync', '_parent', '_id',
+                       '_cache', 'backend_kwargs')
```

Comment:
Don't add a top-level protected key without an underscore. Also, is this missing the key `backend`? I see it's defined in the class `JSONCollection`. Please verify that you're not missing anything by carefully reviewing the output of `dir(cls)` for the synced collections and subclasses.
```suggestion
                       '_cache', '_backend_kwargs')
```